src/internal/mod.rs

```diff
@@ -520,6 +560,30 @@ pub(crate) mod test {
         })
     }
 
+    #[test]
+    fn serde_devicecontext_roundtrip() {
+        use serde_json;
+        let mut recrypt = recrypt::api::Recrypt::new();
+        let priv_key = recrypt.random_private_key();
+        let dev_keys = recrypt::api::SigningKeypair::from_byte_slice(&[
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 138, 136, 227, 221, 116, 9, 241, 149, 253, 82, 219, 45, 60, 186, 93, 114, 202,
+            103, 9, 191, 29, 148, 18, 27, 243, 116, 136, 1, 180, 15, 111, 92,
+        ])
+        .unwrap();
+        let context = DeviceContext::new(
+            TryFrom::try_from("account_id").unwrap(),
+            22,
+            priv_key.into(),
+            DeviceSigningKeyPair::from(dev_keys),
+        );
+
+        let json = serde_json::to_string(&context);
+
+        let de: DeviceContext = serde_json::from_str(&json.unwrap()).unwrap();
+        assert_eq!(context.account_id(), de.account_id());
```

Comment:
It seems like asserting something that's a little more complex than just the account ID here might be good. The (de)serializing of the keys is what I think is more important to test.

Reply:
Yep. Fair enough. It's annoying that we can't derive `PartialEq` because many of the underlying recrypt types don't support it. I am now asserting about both key types.